src/actions/heap/heap.ts

```diff
@@ -99,23 +105,45 @@ export class HeapAction extends Hub.Action {
           if (!heapFieldValue) {
             return
           }
-          const requestBody = this.constructBodyForRequest(
-            request.params.heap_env_id!,
-            heapField,
-            heapFieldValue,
-            properties,
-          )
-          req.post({
-            uri: requestUrl,
-            headers: { "Content-Type": "application/json" },
-            body: requestBody,
-          })
+          requestBatch.push({ heapFieldValue, properties })
+          if (requestBatch.length === HeapAction.ROWS_PER_BATCH) {
+            const requestBody = this.constructBodyForRequest(
+              request.params.heap_env_id!,
+              heapField,
+              requestBatch,
+            )
+            req.post({
```

Comment:
how can we skip `await`ing this? if an error is thrown how does it bubble up?

Reply:
the `onRow` and `onFields` callbacks are not `await`ed in the `streamJsonDetail` implementation so I'm afraid there's no way to `await` this request